draft-js-dnd-plugin/src/modifiers/onDropFile.js

```diff
@@ -0,0 +1,115 @@
+import AddBlock from './addBlock';
+import ModifyBlockData from './modifyBlockData';
+import { Entity } from 'draft-js';
+
+function readFile(file) {
+  return new Promise(resolve => {
+    const reader = new FileReader();
+
+    // This is called when finished reading
+    reader.onload = e => {
+      // Return an array with one image
+      resolve({
+        // These are attributes like size, name, type, ...
+        ...file,
+
+        // This is the files content as base64
+        src: e.target.result,
+
+        // No URL, since nothing on server
+        url: null,
+      });
+    };
+
+    reader.readAsDataURL(file);
+  });
+}
+
+function readFiles(files) {
+  return Promise.all(files.map(readFile));
+}
+
+function getBlockWhereEntityData(state, query) {
+  const map = state.getCurrentContent().get('blockMap').toArray();
+  for (let i = 0; i < map.length; i++) {
+    const block = map[i];
+    const entityData = block.getEntityAt(0) ? Entity.get(block.getEntityAt(0)).getData() : null;
+    if (block.get('type') === 'image' && entityData && query(entityData)) {
+      return block.get('key');
+    }
+  }
+
+  return null;
+}
+
+export default function onDropFile(config) {
+  return function onDropFileInner(e) {
+    const { props, selection, files, editorState, onChange } = e;
+
+    // Get upload function from config or editor props
+    const upload = config.upload || props.upload;
+    const progress = config.progress || props.progress || ((x) => config.emitter.emit('progress', x));
```

Comment:
See my comment on your `How does addListener work? What's the use-case?` question ;)